### PR TITLE
#97 Add scala 2.13 support

### LINKED ISSUE #97: Add support for Scala 2.13
## Feature
It has been requested to be able to use the library in a project built for on Scala 2.13

### PR DESCRIPTION
**Pull request by Zejnilovic**
Closes #97

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -17,15 +17,16 @@ ThisBuild / organization := "za.co.absa"
 
 lazy val scala211 = "2.11.12"
 lazy val scala212 = "2.12.12"
```
Comment (yruslan) [inline build.sbt:19]: There is a newer version of Scala 2.12, if you want to use it.
```suggestion
lazy val scala212 = "2.12.18"
```

**Review by yruslan — APPROVED** [review]
Looks great! (just checked the code)

JDKCollectionConvertersCompat is some piece of magic 😄

File: build.sbt (hunk 1 above)
Comment (Zejnilovic) [reply build.sbt]: Updated

**Review by yruslan — DISMISSED** [review]

**Review by benedeki — CHANGES_REQUESTED** [review]
There is a part I want to look at in more detail, so I apologize for blocking the merge for now (you can override it, if in need 😉  )

File: spark-commons-test/src/main/scala/za/co/absa/spark/commons/test/JDKCollectionConvertersCompat.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spark.commons.test
+
+// Solution based on conversation in scala-collection-compat
+// https://github.com/scala/scala-collection-compat/issues/208
+object JDKCollectionConvertersCompat {
```
Comment (benedeki) [inline spark-commons-test/src/main/scala/za/co/absa/spark/commons/test/JDKCollectionConvertersCompat.scala:21]: While this is pretty clever it's still really hard to understand how it works.
Wouldn't it be better to use shims like [spark-hats](https://github.com/AbsaOSS/spark-hats/tree/master/src/main) or [slick ](https://github.com/slick/slick/tree/v3.3.3/slick/src/main)use.

File: spark-commons-test/src/main/scala/za/co/absa/spark/commons/test/YarnSparkConfiguration.scala [hunk 3]
```diff
@@ -23,7 +23,9 @@ import org.apache.spark.SparkConf
 import org.apache.spark.sql.SparkSession
 import za.co.absa.spark.commons.test.YarnSparkConfiguration._
 
-import scala.collection.JavaConverters.iterableAsScalaIterableConverter
+//import scala.collection.JavaConverters.iterableAsScalaIterableConverter
```
Comment (benedeki) [inline spark-commons-test/src/main/scala/za/co/absa/spark/commons/test/YarnSparkConfiguration.scala:26]: Dead code

File: spark-commons-test/src/main/scala/za/co/absa/spark/commons/test/JDKCollectionConvertersCompat.scala (hunk 2 above)
Comment (Zejnilovic) [reply spark-commons-test/src/main/scala/za/co/absa/spark/commons/test/JDKCollectionConvertersCompat.scala:21]: Happy to go either way. I just used a way they did it in [scala-collection-compat](https://github.com/scala/scala-collection-compat) but at the same time, I tried avoiding adding a new dependency.

**Review by benedeki — APPROVED** [review]
Thank you

**Pull request merged**